### PR TITLE
split cosmology classes from realizations

### PR DESCRIPTION
**Pull request by nstarman**
### Description

Part of #10673

All this PR does is separate the class definitions of a cosmology with its realization(s).
As both the classes and realizations are intended to be accessed from the top-level namespace of the ``cosmology`` package, this PR should have no/minimal/very-easily-fixed impact on users.
Looking forward, this separation will make it easier to add separate entry points for both the classes and realizations so that 3rd party packages can augment``cosmology``'s offerings.

### CODE REVIEW

**Comment by nstarman** [comment]
@pllim, can this be milestoned to 4.3? I just want to get all the tests passing. I've found it encourages followup.

**Comment by nstarman** [comment]
@embray @mhvk @pllim . As the most recent reviewers of a cosmology PR, I thought you'd be the most appropriate to "@" for this one.

**Comment by pllim** [comment]
I can do general review but unfortunately I am uncomfortable with doing a detail review of the science content.

**Comment by nstarman** [comment]
> I can do general review
Much appreciated!

**Comment by nstarman** [comment]
@embray @mhvk @pllim. All tests passing and codecov above the 90% mark (it would be 100% but I replace some never-used code with a warning.)

**Comment by nstarman** [comment]
@embray @mhvk @pllim. Switched change-log to Towncrier news fragment. Test failures unrelated.

**Comment by pllim** [comment]
Thanks for the ping. Unfortunately, I am not qualified to review this, so I'll have to recuse myself.

File: astropy/cosmology/realizations.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            Neff=par["Neff"],
+            m_nu=u.Quantity(par["m_nu"], u.eV),
+            name=key,
+            Ob0=par["Ob0"],
+        )
+        docstr = "{} instance of FlatLambdaCDM cosmology\n\n(from {})"
+        cosmo.__doc__ = docstr.format(key, par["reference"])
+    else:
+        warnings.warn("Please open a PR for your added cosmology realization.")
+        # cosmo = LambdaCDM(
```
Comment (mhvk) [inline astropy/cosmology/realizations.py:39]: Could you add a comment here explaining that for a non-flat LCDM model, it would look like this?

**Review by mhvk — APPROVED** [review]
@nstarman - as you actually promised in the PR, this is really just a technical one that does what I agree is a very logical split -- I'm afraid I had missed that sufficiently to worry this would take me a lot of time, while it turned out to be just 10 minutes.

Anyway, I like this very much - we really should be starting to use entrypoints more -- and I have only one request for a comment about the commented-out code (which I think, contrary to almost anytime else, is worth having here). Approving in the meantime.

File: astropy/cosmology/realizations.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/realizations.py:42]: Thanks. Good suggestion. Done!
```
# For a non-flat LCDM realization, the following is the code necessary
# to create the realization, given the parameters. We comment this out
# as there are not currently any such built-in realizations.
```

**Comment by mhvk** [comment]
Great, thanks.  Merging!

**Pull request merged**